src/main/java/org/jabref/model/entry/BibEntryTypesManager.java

```diff
@@ -127,11 +127,22 @@ public boolean isCustomType(EntryType type, BibDatabaseMode mode) {
         return mode == BibDatabaseMode.BIBLATEX ? BIBLATEX.enrich(type) : BIBTEX.enrich(type);
     }
 
+    public boolean isDifferentCustomOrModifiedType(BibEntryType type, BibDatabaseMode mode) {
+        Optional<BibEntryType> currentlyStoredType = enrich(type.getType(), mode);
+        if (currentlyStoredType.isEmpty()) {
+            // new customization
+            return true;
+        } else {
+            // different customization
+            return !EntryTypeFactory.isEqualNameAndFieldBased(type, currentlyStoredType.get());
+        }
+    }
+
     /**
      * This class is used to specify entry types for either BIBTEX and BIBLATEX.
      */
     static class InternalEntryTypes {
-        private final SortedSet<BibEntryType> customizedTypes = new TreeSet<>();
+        private final SortedSet<BibEntryType> customOrModifiedType = new TreeSet<>();
```

Comment:
you might need to pass a comparator here as well

Reply:
BibEntryType implements comparable, so no comparator needed